distribution/docs/src/main/resources/_contents/_processing-framework/processing-framework-intro-contents.adoc

```diff
@@ -0,0 +1,172 @@
+|             |c369    <ProcessResourceItem>          |                         |               |
+|             +---------------------------------------+                         |               |
+|             |ProcessResource getProcessResource()   |                         |               |
+|             |Boolean isMetacardModified()           |                         |               |
+|             |                                       |                         |               |
+|             +---------------------------------------+                         |               |
+|                                 ^                                             |               |
+|                                 |                                             |               |
+|               +-----------------+-------------+                               |               |
+|               |                               |                               |               |
+|+-----------------------------+ +-----------------------------+ +-----------------------------+|
+||c369 <ProcessCreateItem>     | |c369 <ProcessUpdateItem>     | |c369 <ProcessDeleteItem>     ||
+|+-----------------------------+ +-----------------------------+ +-----------------------------+|
+||                             | |Metacard getOldMetacard()    | |                             ||
+||                             | |                             | |                             ||
+||                             | |                             | |                             ||
+|+-----------------------------+ +-----------------------------+ +-----------------------------+|
++-----------------------------------------------------------------------------------------------+
+....
+
+==== ProcessingFramework
+
+The `ProcessingFramework` is responsible for processing incoming ``ProcessRequest``s that contain a `ProcessItem`. A `ProcessingFramework` should never block. It receives
+its ``ProcessRequest``s from a `PostIngestPlugin` on all CUD operations to the Catalog. In order to determine whether or not asynchronous processing
+is required by the `ProcessingFramework`, the `ProcessingFramework` should mark any request it has submitted back the Catalog, otherwise a processing loop may occur.
+For example, the default *In-Memory Processing Framework* adds a `POST_PROCESS_COMPLETE` flag to the Catalog CUD request after processing. This flag is checked by the
+`ProcessingPostIngestPlugin` before a `ProcessRequest` is sent to the `ProcessingFramework`. For an example of a `ProcessingFramework`, please refer to the `org.codice.ddf.catalog.async.processingframework.impl.InMemoryProcessingFramework`.
+
+==== ProcessRequest
+
+Contains a list of ``ProcessItem``s for the `ProcessingFramework` to process. Once a `ProcessRequest` has been processed by a `ProcessingFramework`, the `ProcessingFramework`
```

Comment:
Start this sentence with "A `ProcessRequest` contains" to be consistent with the other sections